### DISCUSSION TITLE
How to pass multiple codes to `--ignored-error-codes`...

### DISCUSSION THREAD

**Discussion by jmhickman** [foundry-rs/foundry#3500] [Help]
`forge test --ignored-error-codes 2072,6321,5667,2018 --match-path test/Test.t.sol`
`forge test --ignored-error-codes "2072,6321,5667,2018" --match-path test/Test.t.sol`
`forge test --ignored-error-codes "2072, 6321, 5667, 2018" --match-path test/Test.t.sol`

All of these result in the same error, just with formatting differences:
```
error: Invalid value "2072,6321,5667,2018" for '--ignored-error-codes <ERROR_CODES>': invalid digit found in string

For more information try --help
```

The book says it's a comma-separated list of values, but I'm not seeing that working.

Anyone know what the magic incantation is?

**Comment by jmhickman** [comment]
Honestly, I would love a "suppress all warnings"

**Comment by Genysys** [comment]
from here , the flag expects a vector (array) of numbers , so I would
remove the quotes around them


https://github.com/foundry-rs/foundry/commit/d586f3202639e2e0499e9a766640d1c06fbf1243#diff-fe8496abbb64c2973db10f0703234d17eb9d104f211155751b00372dd22cdbfcR56

**Reply by jmhickman** [reply]
Unfortunately, that isn't it.

```
forge test --ignored-error-codes 2072, 6321, 5667, 2018 --match-path test/Test.t.sol -vvv
error: Found argument '6321,' which wasn't expected, or isn't valid in this context

USAGE:
    forge test [OPTIONS]

For more information try --help

forge test --ignored-error-codes 2072,6321,5667,2018 --match-path test/Test.t.sol -vvv
error: Invalid value "2072,6321,5667,2018" for '--ignored-error-codes <ERROR_CODES>': invalid digit found in string

For more information try --help
```

**Comment by DaniPopes** [comment]
You can use:
`forge test --ignored-error-codes 2072 --ignored-error-codes 6321 --ignored-error-codes 5667 --ignored-error-codes 2018`

or add them to your project's `foundry.toml`:
`ignored_error_codes=[2072, 6321, 5667, 2018]`

> Honestly, I would love a "suppress all warnings"

I think this would be a good addition, can you create an issue for it?

**Reply by jmhickman** [reply]
The toml addition got me over the finish line, so thanks for that!

Is the description of how `--ignore-error-codes` works erroneous? Or is the current behavior a bug?

**Reply by DaniPopes** [reply]
The description is wrong for how it works right now, but we could add support for comma-separated list parsing CC @mattsse

**Reply by mattsse** [reply]
+1 on supporting multiple values as args here and or suppress all

**Reply by jmhickman** [reply]
I'm in the middle of audit work at the moment, but I can make an issue in a few days.
